core/src/main/java/com/microsoft/applicationinsights/internal/channel/samplingV2/FixedRateSamplingTelemetryProcessor.java

```diff
@@ -0,0 +1,212 @@
+     */
+    public FixedRateSamplingTelemetryProcessor() {
+        this.samplingPercentage = 100.00;
+        this.includedTypes = new HashSet<Class>();
+        this.excludedTypes = new HashSet<Class>();
+        try {
+            this.allowedTypes = new HashMap<String, Class>() {{
+                put(dependencyTelemetryName, Class.forName("com.microsoft.applicationinsights.telemetry.RemoteDependencyTelemetry"));
+                put(eventTelemetryName, Class.forName("com.microsoft.applicationinsights.telemetry.EventTelemetry"));
+                put(exceptionTelemetryName, Class.forName("com.microsoft.applicationinsights.telemetry.ExceptionTelemetry"));
+                put(pageViewTelemetryName, Class.forName("com.microsoft.applicationinsights.telemetry.PageViewTelemetry"));
+                put(requestTelemetryName, Class.forName("com.microsoft.applicationinsights.telemetry.RequestTelemetry"));
+                put(traceTelemetryName, Class.forName("com.microsoft.applicationinsights.telemetry.TraceTelemetry"));
+            }};
+        } catch (ClassNotFoundException e) {
+            InternalLogger.INSTANCE.trace("Unable to locate telemetry classes");
+        }
+    }
+
+    /**
+     * This method returns a set of classes of excluded types specified by user
+     *
+     * @return
+     */
+    public Set<Class> getExcludedTypes() {
+        return excludedTypes;
+    }
+
+    /**
+     * This method returns a set of classes of included types specified by user
+     *
+     * @return
+     */
+    public Set<Class> getIncludedTypes() {
+        return includedTypes;
+    }
+
+
+    private void setIncludedOrExcludedTypes(String value, Set<Class> typeSet) {
+
+        if (!StringUtils.isNullOrEmpty(value)) {
+            value = value.trim();
+            if (!StringUtils.isNullOrEmpty(value) && allowedTypes.containsKey(value)) {
+                typeSet.add(allowedTypes.get(value));
+            } else {
+                InternalLogger.INSTANCE.error("Item is either not allowed to sample or is empty");
+            }
+        } else {
+            InternalLogger.INSTANCE.error("Empty types cannot be considered");
+        }
+    }
+
+    /**
+     * Gets the sample rate currently set
+     */
+    double getSamplingPercentage() {
+        return samplingPercentage;
+    }
+
+    /**
+     * Sets the user defined sampling percentage
+     *
+     * @param samplingPercentage
+     */
+    public void setSamplingPercentage(String samplingPercentage) {
+        this.samplingPercentage = Double.valueOf(samplingPercentage);
+    }
+
+    /**
+     * This method determines if the telemetry needs to be sampled or not.
+     *
+     * @param telemetry
+     * @return
+     */
+    @Override
+    public boolean process(Telemetry telemetry) {
+
+        double samplingPercentage = this.samplingPercentage;
+
+        if (telemetry instanceof SupportSampling) {
+
+            if (isSamplingApplicable(telemetry.getClass())) {
+
```

Comment:
what's the deal with so many empty lines?

Reply:
It was just to make code cleaner to look. we can remove them.